lib/default.nix

```diff
@@ -204,6 +204,18 @@ let
         ) (lib.attrsToList hosts)
       );
 
+      # Check if the given module is wrapped in a function expecting "publisher" as a named argument,
+      # and - if so - call that function with the current flake. This enables us to reference
+      # the publishing flake when re-using exported modules in consuming, downstream flakes.
+      withPublisher =
+        module:
+        let
+          module' = if builtins.isString module || builtins.isPath module then import module else module;
```

Comment:
One downside with importing a module manually is that its result doesn't get annotated with its "_file" attribute. Which is used by the module system to display errors in the right location.

Fixing this is a bit tricky as it happens during the module evaluation.

Reply:
Yes, but this took me a while, but the way it works here is that we just import `module'` (prime) to check whether that particular module is wrapped or not.

The actual import of `module` (no prime) happens in the `lib.modules.importApply` call which handles `_file` for us. i.e.:

```
nix-repl> nixosModules.common._file
"/nix/store/7ds7wj847q654jqc5gs29zyfa8gzph08-source/modules/nixos/common.nix"
```